cmd/system-probe/probe_test.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"net/http/httptest"
+	"testing"
+
+	"github.com/DataDog/datadog-agent/pkg/ebpf"
+	"github.com/DataDog/datadog-agent/pkg/ebpf/netlink"
+	"github.com/DataDog/datadog-agent/pkg/process/util"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDecode(t *testing.T) {
+	rec := httptest.NewRecorder()
+
+	in := &ebpf.Connections{
+		Conns: []ebpf.ConnectionStats{
+			{
+				Source: util.AddressFromString("10.1.1.1"),
+				Dest:   util.AddressFromString("10.1.1.1"),
+				SPort:  1000,
+				DPort:  9000,
```

Comment:
nit: can we set all the fields to non-zero values ? (just to ensure that decoding worked for those) and also have different source and dests values

Reply:
Done